app/services/xccdf_tailoring_file.rb

```diff
@@ -46,11 +46,30 @@ def profile_builder(xml)
   end
 
   def rule_selections_builder(xml)
-    @rule_ref_ids.each do |rule_ref_id, selected|
+    resolved_rules.each do |rule_ref_id, selected|
       xml[XCCDF].select(idref: rule_ref_id, selected: selected)
     end
   end
 
+  def resolved_rules
+    @rule_ref_ids.select { |_id, selected| selected }.each do |rule_ref_id, _selected|
+      rule_for(rule_ref_id)&.left_rule_group_relationships&.each do |rgr|
+        @rule_ref_ids[rule_ref_id] = false if deselect?(rgr)
```

Comment:
Is `@rule_ref_ids[rule_ref_id]` always `true` by default?
Shouldn't we explicitly assign `@rule_ref_ids[rule_ref_id] = true` in case if `deselect?` is `false`?

Reply:
Sorry, I missed this yesterday! `@rule_ref_ids[rule_ref_id]` is initially set to either true or false when it's passed into the initialize method in this file so if we don't set it to false at this line, it'll just keep its original value.